### ISSUE TITLE
Improve clarity around auth errors in serviceloadbalancer

### ISSUE THREAD

**Issue by sandromello** [kubernetes/contrib#478]
I'am having some trouble starting haproxy.

Kubernetes Server Version: 1.1
Docker Version: 1.9.1 / Api: 1.21
Master/Minion OS: CoreOS - 899.6.0

This problem is related to:
https://github.com/kubernetes/kubernetes/issues/19441

When I started a load balancer replication container:

``` bash
kubectl create -f lb-rc.yaml
```

The haproxy process does not starts.

``` yaml
# lb-rc.yaml
apiVersion: v1
kind: ReplicationController
metadata:
  name: service-loadbalancer
  labels:
    app: service-loadbalancer
    version: v1
spec:
  replicas: 1
  selector:
    app: service-loadbalancer
    version: v1
  template:
    metadata:
      labels:
        app: service-loadbalancer
        version: v1
    spec:
      nodeSelector:
        role: loadbalancer
      containers:
      - image: gcr.io/google_containers/servicelb:0.2
        imagePullPolicy: Always
        name: haproxy
        ports:
        # All http services
        - containerPort: 80
          hostPort: 80
          protocol: TCP
        resources: {}
```

The container has no ports associate with haproxy

``` bash
$ kubectl exec service-loadbalancer-7ke2x -it -- netstat -ltpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 :::8081                 :::*                    LISTEN      1/service_loadbalan
```

Executing `./haproxy_reload` shows me the error

``` bash
$ kubectl exec service-loadbalancer-7ke2x -it -- ./haproxy_reload
cat: can't open '/var/run/haproxy.pid': No such file or directory
[ALERT] 039/183101 (39) : parsing [/etc/haproxy/haproxy.cfg:32] : 'listen' cannot handle unexpected argument ':1936'.
[ALERT] 039/183101 (39) : parsing [/etc/haproxy/haproxy.cfg:32] : please use the 'bind' keyword for listening addresses.
[ALERT] 039/183101 (39) : Error(s) found in configuration file : /etc/haproxy/haproxy.cfg
[WARNING] 039/183101 (39) : config : proxy 'stats' has no 'bind' directive. Please declare it as a backend if this was intended.
[WARNING] 039/183101 (39) : config : missing timeouts for proxy 'stats'.
   | While not properly invalid, you will certainly encounter various problems
   | with such a configuration. To fix this, please ensure that all following
   | timeouts are set to a non-zero value: 'client', 'connect', 'server'.
[WARNING] 039/183101 (39) : config : log format ignored for frontend 'httpfrontend' since it has no log address.
[ALERT] 039/183101 (39) : Fatal errors found in configuration.
error: error executing remote command: Error executing command in container: Error executing in Docker Container: 1
```

Checking the file /etc/haproxy/haproxy.cfg:

```
$ kubectl exec service-loadbalancer-7ke2x -it -- cat /etc/haproxy/haproxy.cfg |grep -C 1 listen
# haproxy stats, required hostport and firewall rules for :1936
listen stats :1936
    mode http
```

The /template.cfg is correctly configured when I start the service-loadbalancer container:

``` bash
$ kubectl exec service-loadbalancer-7ke2x -it -- cat /template.cfg |grep -C 1 listen
# haproxy stats, required hostport and firewall rules for :1936
listen stats
    bind *:1936
```

Does the image gcr.io/google_containers/servicelb:0.2 is updated correctly or I'am missing something?

**Comment by sandromello** [comment]
I think my cluster is wrongly configured. I'am using other servicelb image and the config never rewrites. I'll dig further into that.

**Comment by sandromello** [comment]
Resolved. I had a wrong configuration in services that prevented from reaching to the apiserver. After that I've got some problems with the service-loadbalancer connecting to apiserver:

```
handlers.go:37] Unable to authenticate the request due to an error: crypto/rsa: verification error\n 
```

I've deleted the default secret from my default namespace and the haproxy.cfg started being rewrited accordingly. 

Maybe logging communication errors with apiserver into service-loadbalancer container will facilitate troubleshooting related problems.
